src/promptflow/promptflow/_utils/dataclass_serializer.py

```diff
@@ -115,3 +115,13 @@ def assertEqual(a: dict, b: dict, path: str = ""):
             assertEqual(a[i], b[i], path + f"[{i}]")
     else:
         assert a == b, f"{path}: {a} != {b}"
+
+
+# Since the output of eager flow may not be a dict, we need to convert it to a dict in batch mode.
```

Comment:
We may add some examples here to help understand the behavior.

Reply:
Added.